10up/insecure-content-warning
# Add support for the File block

### Description of the Change

Currently we only look for insecure items in the `src` or `srcset` attributes. The core File block has a preview option that will embed a preview using the `object` element. This stores the URL under the `data` attribute which never gets looked at.

This PR fixes that by checking the element being scanned and if it is an `object`, ensure we process that correctly. It also adds handling for these elements in the Classic Editor, if someone were to add a File block in Gutenberg and then edit that item in the Classic Editor.

And finally, it ensures we replace all insecure URLs instead of just finding the first match. This isn't an issue when dealing with images but with the structure of the File block, the URL is used multiple times.

### Alternate Designs

None

### Benefits

Insecure URLs will be flagged when used in the File block

### Possible Drawbacks

None

### Verification Process

1. Add a File block
2. Add an insecure file to that block (like a PDF)
3. Save the post
4. Ensure the file URL is getting flagged

### Checklist:


### Changelog Entry

> Fix - add support for the core File block
